### PR TITLE
feat: added basic implementation and skeleton for Round Start mechanism

### PR DESCRIPTION
**Pull request by DedsecKnight**
TODO: 

- [x] merge and resolve conflicts with #23 
- [x] added problem generation and visibility logic on round creation
- [x] resolves remaining TODOs

### CODE REVIEW

File: central-service/services/problem.go [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
 
 type ProblemService struct {
 	db *gorm.DB
 }
 
+type DifficultyParameter struct {
+	NumEasyProblems int
+	NumMediumProblems int
+	NumHardProblems int
+}
```
Comment (Punchwood2003) [inline central-service/services/problem.go:18]: Is this something that we intend to leave up to the users to configure, or is this something that is going to be unmodifiable? Just curious what our intent is regarding the ability to allow users to choose the number of problems they wish to have.

**Review by Punchwood2003 — APPROVED** [review]
A lot of really solid work that looks pretty great to me! Loving the abstraction for the sake of other internal services that might need to use these components without wanting to worry about how it is implemented 👍

File: central-service/services/problem.go (hunk 1 above)
Comment (DedsecKnight) [reply central-service/services/problem.go:18]: This is something that we would leave up to the users to configure, but adding a default value could be a great idea too.

File: central-service/models/round.go [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
 type Round struct {
-	ID       uint `gorm:"primaryKey" json:"id"`
-	Duration int
-	RoomID   uuid.UUID
+	ID              uint `gorm:"primaryKey" json:"id"`
+	LastUpdatedTime time.Time
+	Duration        int       `json:"duration"` // Duration in minutes
+	RoomID          uuid.UUID `json:"roomID"`
+	Status          string
+	ProblemSet			[]Problem `gorm:"many2many:round_problems;" json:"problems"`
```
Comment (1solomonwakhungu) [inline central-service/models/round.go]: nit: massive space here

**Review by 1solomonwakhungu — APPROVED** [review]
👍

**Pull request merged**